internal/provider/kubernetes/controller.go

```diff
@@ -332,13 +330,13 @@ func (r *gatewayAPIReconciler) Reconcile(ctx context.Context, _ reconcile.Reques
 				return reconcile.Result{}, err
 			}
 
-			r.log.Error(err, fmt.Sprintf("failed process TLS SecretRef for EnvoyProxy for gatewayClass %s", managedGC.Name))
-			gc := status.SetGatewayClassAccepted(
-				managedGC.DeepCopy(),
```

Comment:
I didn't understand why we do deepcopy here, we post GC with old status to gatewayapi layer in the past.

cc @arkodg